src/gui/qgsmapcanvas.h

```diff
@@ -273,6 +274,25 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
     //! @note added in 2.12
     void setLayerStyleOverrides( const QMap<QString, QString> &overrides );
 
+    /**
+     * Sets a map \a theme to show in the canvas. The theme name must match
+     * a theme present in the associated project's QgsMapThemeCollection.
+     * If an empty string is passed then the current theme association will be
+     * cleared. The layers from the previously associated theme will remain
+     * in the canvas, and a call to setLayers() may be necessary to define
+     * which layers should be shown in the canvas.
```

Comment:
Could we add here that when map canvas is associated to a theme, it will follow its changes and update layers/styles accordingly?